src/javascript/app_2/App/Components/Elements/SettingsDialog/__tests__/language_dialog.spec.js

```diff
@@ -0,0 +1,30 @@
+import React                  from 'react';
+import { expect }             from 'chai';
+import { configure, shallow } from 'enzyme';
+import Adapter                from 'enzyme-adapter-react-16';
+import LanguageDialog         from '../language_dialog.jsx';
+
+configure({ adapter: new Adapter() });
+
+describe('LanguageDialog', () => {
+    it('should render one <LanguageDialog /> component', () => {
+        const wrapper = shallow(<LanguageDialog />);
+        expect(wrapper).to.have.length(1);
+    });
+    it('should have .show when is_visible is true and is_settings_on is true', () => {
+        const wrapper = shallow(<LanguageDialog is_visible={true} is_settings_on={true} />);
+        expect(wrapper.find('.show').exists()).to.be.true;
+    });
+    it('should not have .show when is_visible is true and is_settings_on is false', () => {
+        const wrapper = shallow(<LanguageDialog is_visible={true} is_settings_on={false} />);
+        expect(wrapper.find('.show').exists()).to.be.false;
+    });
+    it('should not have .show when is_visible is false and is_settings_on is true', () => {
+        const wrapper = shallow(<LanguageDialog is_visible={false} is_settings_on={true} />);
+        expect(wrapper.find('.show').exists()).to.be.false;
+    });
+    it('should not have .show when is_visible is false and is_settings_on is false', () => {
+        const wrapper = shallow(<LanguageDialog is_visible={false} is_settings_on={false} />);
+        expect(wrapper.find('.show').exists()).to.be.false;
+    });
```

Comment:
I think we should add one truth-y test as well, just to make sure the data received rendered correctly.

Reply:
there are 4 conditions and just one of them is truthy, that's when both props is_visible and is_settings are true, this case is tested

Reply:
You are right, I was thinking about implementing test cases for `getAllowedLanguages` as well. But that should not be test here, we should write one for `getAllowedLanguages` on it's own specification file.